### PR TITLE
#164275560 Integrate CI and maintainability tool 

### PR DESCRIPTION
**Pull request by Bakley**
#### What does this PR do?
- Integrate Travis CI, Coveralls code coverage service and CodeClimate maintainability service
#### What are the tasks to be completed?
- Dummy tests should run on Travis CI.
- After tests run on Travis CI, the generated coverage report should be sent to coveralls.
- Code climate should be able to analyze code written.
- Displays Travis CI Badge, Coveralls Badge and Code Climate Maintainability Badge on the README.
#### How can this be manually tested?
- After cloning the repo:
`
git checkout ch-integrate-CI-and-maintainability-tool-164275560
`
#### Any background context you want to provide?
- The Travis build passes so if the build fails it will be caused mostly due to failing tests
#### What are the relevant pivotal tracker stories?
#164275560
#### Screenshots (if appropriate)
travis
dummy_test
true_test

code_climate
cov

### CODE REVIEW

**Review by Ogutu-Brian — DISMISSED** [review]

**Review by IssaIan — DISMISSED** [review]

**Comment by IssaIan** [comment]
Try squashing your commits next time.

**Comment by IssaIan** [comment]
Related commits can be squashed into one. This will avoid clustering of commits.

**Comment by Bakley** [comment]
@IssaIan Thank you pointing that out. I will be squashing the commit

**Review by matthenge — APPROVED** [review]

**Review by davenjeru — CHANGES_REQUESTED** [review]
You need to add the command to run a coverage report on the tested files. This coverage report will be sent to coveralls. Currently coverage is unknown because there was no coverage report sent to coveralls. Also, add the coveralls and code-climate reporter to Pull Requests, currently only Travis CI is seen.

File: .travis.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/usr/env bash
```
Comment (davenjeru) [inline .travis.sh]: There is a section on Travis CI that allows one to set environment variables and hence kept secret. These variables should not be committed.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+python:
+  - 3.6
+addons:
+  postgresql: "9.4"
+services: postgresql
+env:
+  - DJANGO=2.1.7 DB=postgresql
+
+before_install:
+  - export DJANGO_SETTINGS_MODULE=config.settings
```
Comment (davenjeru) [inline .travis.yml]: Same for here on exporting environment variables.

File: api/config/settings.py [hunk 3]
```diff
@@ -312,9 +312,9 @@
 MOBILE_ACTIVATION_TOKEN_HASH_ALGORITHM = 'sha256'
 
 # Test Settings
-TEST_PAYLOAD_PATH = str(API_DIR) + '/utils/test/'
-TEST_DATA_PATH = TEST_PAYLOAD_PATH + 'data/'
-TEST_RUNNER = 'utils.test.test_runner.CMTestRunner'
+#TEST_PAYLOAD_PATH = str(API_DIR) + '/utils/test/'
```
Comment (davenjeru) [inline api/config/settings.py]: Avoid committing code that is commented. If the code is not needed, delete it entirely.

**Review by davenjeru — CHANGES_REQUESTED** [review]

File: .travis.sh (hunk 1 above)
Comment (Bakley) [reply .travis.sh]: Found the Environment variables, moving the variables.

File: api/config/settings.py (hunk 3 above)
Comment (Bakley) [reply api/config/settings.py]: Done

File: .travis.yml (hunk 2 above)
Comment (Bakley) [reply .travis.yml]: Done

File: .travis.sh (hunk 1 above)
Comment (Bakley) [reply .travis.sh]: Done

Comment (davenjeru) [inline .travis.sh]: @Bakley, delete this file from the commit. It is no longer needed.

**Review by davenjeru — CHANGES_REQUESTED** [review]

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+before_script:
+  - psql -c "create user django WITH PASSWORD 'password';" -U postgres
+  - psql -c "create database drf;" -U postgres
+
+script:
+- python api/manage.py makemigrations
+- python api/manage.py migrate
+- python api/manage.py test users
+- coverage run api/manage.py test
```
Comment (davenjeru) [inline .travis.yml:27]: Ignore files that are in the `virtualenv` from being calculated for coverage.
image

It is showing over 700 entries. Ignore the files what we did not write code for.

**Review by davenjeru — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 4 above)
Comment (Bakley) [reply .travis.yml:27]: Yes worked on it

File: .travis.sh (hunk 1 above)
Comment (Bakley) [reply .travis.sh]: Deleted

File: .travis.yml (hunk 4 above)
Comment (Bakley) [reply .travis.yml:27]: @davenjeru Check the branch, worked on the feedback

**Review by davenjeru — APPROVED** [review]

**Comment by Ogutu-Brian** [comment]
LGTM

**Review by Ogutu-Brian — APPROVED** [review]
LGTM

**Pull request merged**